actionview/app/assets/javascripts/rails-ujs/utils/event.coffee

```diff
@@ -66,3 +66,4 @@ Rails.delegate = (element, selector, eventType, handler) ->
     if target instanceof Element and handler.call(target, e) == false
       e.preventDefault()
       e.stopPropagation()
+  , true
```

Comment:
the linked blog post suggests `{ capture: true }`, is there a difference between that and just `true` here?

Reply:
[These](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) indicate that they're the same.

Reply:
Cool, thanks for confirming.

So if I understand correctly, this change will allow events fired by React 17 to be handled by Rails UJs, same as they were in previous versions of React. And that would apply for the all the events [here](https://github.com/rails/rails/blob/c5184a747430d833bd3f0e54254fc48f2b164007/actionview/app/assets/javascripts/rails-ujs/start.coffee#L34..LL68).

Are there any risks for non-React users?

Reply:
Yes, that's correct. As far as risks for non-React users I'm noticing that there are some failing tests surrounding event bubbling. I started to look into them but I'm not all that familiar with @rails/ujs and/or the test suite.

Reply:
With a fair bit of tinkering, I got the test results. Will have a think about how to get them to show in CI. For now, here is the qunit output: https://gist.github.com/ghiculescu/02b72bf2826615201db8811a0e04f521

So I think these are the failing tests:

- data-confirm: clicking on the children of a disabled button should not trigger a confirm. (1, 0, 1)
- data-remote: form's submit bindings in browsers that don't support submit bubbling (2, 2, 4)
- data-remote: returning false in form's submit bindings in non-submit-bubbling browsers (1, 0, 1)

Reply:
For `data-confirm` the functionality is working but the test is not. Since capture propagates events from the outermost element in when the event on the disabled button fires we call [`stopImmediatePropagation`](https://github.com/rails/rails/blob/main/actionview/app/assets/javascripts/rails-ujs/features/disable.coffee#L7) which stops the click event from firing on the child as well.

I would imagine the other specs are failing for a similar reason but I haven't been able to narrow it down.

Reply:
So I _think_ the other specs are also failing because @rails/ujs is calling [`stopImmediatePropagation`](https://github.com/rails/rails/blob/main/actionview/app/assets/javascripts/rails-ujs/features/remote.coffee#L72) which is causing the events in the spec to not be called. I can't tell if this is actually breaking functionality or if the specs are testing too close to the implementation.

Reply:
Is there a way to write the specs better to match this new behaviour?

Reply:
Hmm I tried seeing if the behavior was actually broken by testing some of the scenarios in a real application. It appears that things are indeed not quite working as intended. I think the issue is @rails/ujs calls stopPropagation in various places, this was fine with bubbling but with capture it's having some unintended consequences. I feel like in order to enable `useCapture` a bit of a larger refactor might be in order.